authors/apps/articles/serializers.py

```diff
@@ -0,0 +1,51 @@
+from rest_framework import serializers
+from .models import Article
+from rest_framework.fields import CurrentUserDefault
+from authors.apps.profiles.serializers import ProfileSerializer
+
+import re
+
+
+class ArticleSerializer(serializers.ModelSerializer):
+    """
+    Defines the article serializer
+    """
+    title = serializers.CharField(required=True)
+    body = serializers.CharField(required=True)
+    description = serializers.CharField(required=True)
+    slug = serializers.SlugField(required=False)
+    image_url = serializers.URLField(required=False)
+    created_at = serializers.DateTimeField(read_only=True)
+    updated_at = serializers.DateTimeField(read_only=True)
+    author = ProfileSerializer(read_only=True)
+
+    class Meta:
+        model = Article
+        fields = ['title', 'slug', 'body',
+                  'description', 'image_url', 'created_at', 'updated_at', 'author']
+
+    def create(self, validated_data):
+        return Article.objects.create(**validated_data)
+
+    def validate(self, data):
+        # The `validate` method is used to validate the title,
+        # description and body
+        # provided by the user during creating or updating an article
+        title = data.get('title', None)
+        description = data.get('description', None)
+
+        # Validate title is not a series of symbols or non-alphanumeric characters
+        if re.match(r"[!@#$%^&*~\{\}()][!@#$%^&*~\{\}()]{2,}", title):
+            raise serializers.ValidationError(
+                """A title must not contain two symbols/foreign characters following each other"""
+            )
+        # Validate the description is not a series of symbols or
+        # non-alphanumeric characters
+        if re.match(r"[!@#$%^&*~\{\}()][!@#$%^&*~\{\}()]{2,}", description):
+            raise serializers.ValidationError(
+                """A description must not contain two symbols/foreign characters following each other"""
```

Comment:
Indent this

Reply:
I am not sure what you suggest I do about this line